contracts/scripts/deploy.sh

```diff
@@ -105,6 +105,43 @@ function deploy_and_instrument_game {
   eval "$__resultvar='$contract_address'"
 }
 
+function instrument_marketplace {
+  local  __resultvar=$1
+  local code_hash=$2
+  local contract_name=$3
+  local salt=$4
+  local ticket_token=$5
+  local game_token=$6
+  local game=$7
+
+  # --- CREATE AN INSTANCE OF THE CONTRACT
+
+  cd "$CONTRACTS_PATH"/marketplace
+
+  local blocks_per_hour=3600
+  local initial_price="$TOTAL_BALANCE"
+  local min_price=1
+  local sale_price_multiplier=2
+
+  local contract_address
+  contract_address=$(cargo contract instantiate --url "$NODE" --constructor new \
+    --args "$ticket_token" "$game_token" "$initial_price" "$min_price" "$sale_price_multiplier" "$blocks_per_hour" \
+    --suri "$AUTHORITY_SEED" --salt "$salt")
+  contract_address=$(echo "$contract_address" | grep Contract | tail -1 | cut -c 15-)
+
+  echo "Marketplace for $contract_name instance address: $contract_address"
+
+  # --- GRANT PRIVILEGES ON THE CONTRACT
+
+  cd "$CONTRACTS_PATH"/access_control
+
+  cargo contract call --url "$NODE" --contract "$ACCESS_CONTROL" --message grant_role --args "$AUTHORITY" 'Owner('"$contract_address"')' --suri "$AUTHORITY_SEED"
+  cargo contract call --url "$NODE" --contract "$ACCESS_CONTROL" --message grant_role --args "$AUTHORITY" 'Admin('"$contract_address"')' --suri "$AUTHORITY_SEED"
+  cargo contract call --url "$NODE" --contract "$ACCESS_CONTROL" --message grant_role --args "$game" 'Admin('"$contract_address"')' --suri "$AUTHORITY_SEED"
```

Comment:
why is the game contract made an Admin?

Reply:
My idea here was that the game contract is going to call `reset()` after it puts in the tokens.